src/cluster.c

```diff
@@ -1895,19 +1893,32 @@ void sflushCommand(client *c) {
      * a new slotRangeArray. It is allocated on heap since there is a chance
      * that FLUSH SYNC will be running as blocking ASYNC and only later reply
      * with slot ranges */
+    unsigned char slots_to_flush[CLUSTER_SLOTS] = {0}; /* Requested slots to flush */
     slotRangeArray *myslots = NULL;
     for (int i = 0; i < slot_ranges->num_ranges; i++) {
-        slotRange *sr = &slot_ranges->ranges[i];
-        for (int slot = sr->start; slot <= sr->end; slot++)
-            if (clusterIsMySlot(slot))
-                myslots = slotRangeArrayAppend(myslots, slot);
+        for (int j = slot_ranges->ranges[i].start; j <= slot_ranges->ranges[i].end; j++) {
+            if (clusterIsMySlot(j)) {
+                myslots = slotRangeArrayAppend(myslots, j);
+                slots_to_flush[j] = 1;
+            }
+        }
     }
-    zfree(slot_ranges);
 
-    if (myslots == NULL) {
-        addReply(c, shared.emptyarray);
+    /* Verify that all slots of mynode got covered. See sflushCommand() comment. */
+    int all_slots_covered = 1;
```

Comment:
I'm reverting the logic of SFLUSH back to the original way in unstable. It was expecting the command to cover all the slots of the node so it can trigger "async flushall". We won't be able to flush partial slots asynchronously, so I guess we should not introduce any regression.